File: backend/booking-api/src/main/java/com/lhsystems/booking/api/dto/BookingResponseDTO.java
package com.lhsystems.booking.api.dto;

import java.time.Instant;

public record BookingResponseDTO(
        String bookingId,
        String pnr,
        String status,
        FlightDTO flight,
        PassengerDTO passenger,
        Instant createdAt,
        Boolean cancellable) {}

File: frontend/src/api/models/bookingResponseDTO.ts
import type { BookingResponseDTOStatus } from "./bookingResponseDTOStatus";
import type { FlightDTO } from "./flightDTO";
import type { PassengerDTO } from "./passengerDTO";

export interface BookingResponseDTO {
  bookingId?: string;
  /** Amadeus PNR record locator */
  pnr?: string;
  status?: BookingResponseDTOStatus;
  flight?: FlightDTO;
  passenger?: PassengerDTO;
  createdAt?: string;
  /** Whether the booking can still be cancelled (>24h before departure) */
  cancellable?: boolean;
}
